keps/sig-cli/2985-public-krm-functions-registry/README.md

```diff
@@ -0,0 +1,938 @@
+central index repository owned by CNCF.
+
+#### Prior Art
+
+Similar model is used by [Krew Index](https://github.com/kubernetes-sigs/krew-index)
+(kubectl’s plugin index) and [Terraform Registry](https://registry.terraform.io/)
+(a public registry for Terraform modules).
+
+In Krew Index, only the manifest files (plugin metadata) are published to the
+[krew-index](https://github.com/kubernetes-sigs/krew-index) repo.
+[Here](https://github.com/kubernetes-sigs/krew-index/blob/master/plugins/access-matrix.yaml)
+is an example plugin manifest file.
+
+In [Terraform Registry](https://registry.terraform.io/), when contributors want
+to publish a module, they will need to allow Terraform Registry to register a
+webhook in their public GitHub repository to detect git tags as releases.
+
+#### Pros and Cons
+
+Pros:
+
+* Scale well. Maintainers only need to review function metadata files when
+  contributors publish their functions.
+* Encourage the KRM functions community to grow larger and faster.
+* It’s possible to support binaries as KRM functions in the registry, since they
+  are built and released by the publishers.
+
+Cons:
+
+* Quality (e.g. test coverage) of the code is hard to be enforced.
+* Security may be a concern if we want to support binaries as KRM functions in
+  the registry.
+
+### Mixture Model
+
+We can mix the 2 management models above.
+
+We can manage the source code of the Kustomize provided KRM functions in-tree.
+Generic KRM functions like set-labels, set-annotations and set-namespace can be
+included. All the KRM functions provided by kustomize must go through a security
+audit.
+
+The model of centralized index with distributed release is more flexible and
+more suitable for all vendors and other contributors. We can manage the source
+code of the functions contributed by the community out-of-tree.
+
+All KRM functions in the registry must provide the metadata in the repo. We must
+standardize the metadata format for KRM functions, since we will require all
+contributors to follow it. A website can be built using the metadata information.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+Simple smoke tests can be set up to ensure the functions are compliant with the
+[KRM functions spec].
```

Comment:
Which functions would these run against? If its for any non-first-party, what happens when they start failing? What are the considerations for security of using SIG infra for such tests?

Reply:
> What are the considerations for security of using SIG infra for such tests?

This is a good point. I have updated it.